example/src/main/java/com/vimeo/example/MainActivity.kt

```diff
@@ -0,0 +1,129 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.vimeo.example
+
+import android.content.Intent
+import android.net.Uri
+import android.os.Bundle
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import com.vimeo.networking2.*
+import com.vimeo.networking2.config.VimeoApiConfiguration
+import kotlinx.android.synthetic.main.activity_main.*
+
+/**
+ * The sample activity in which we create an [Authenticator] instance to authenticate with the server.
+ */
+class MainActivity : AppCompatActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
+        // Construct a configuration instance.
+        val configuration = VimeoApiConfiguration.Builder(CLIENT_ID, CLIENT_SECRET, listOf(ScopeType.PUBLIC))
+            .withCodeGrantRedirectUrl("https://example.com")
+            .build()
+
+        // Initialize the authenticator instance with the configuration and the api client.
+        val authenticator = Authenticator(configuration)
+        val apiClient = VimeoApiClient(configuration, authenticator)
+
+        // If we are handling a code grant redirect, this URI will be present, otherwise it's a normal app start.
+        val redirectUri = intent?.data
+
+        if (redirectUri != null) {
+            // Log in using the code grant URI provided to us by the redirect.
+            authenticator.authenticateWithCodeGrant(redirectUri.toString(), vimeoCallback(
+                onSuccess = {
+                    toast("Successfully logged in.")
+                    loginStatus.text = authenticator.currentAccount.asLoginStatus()
+                },
+                onError = { toast("Unable to log in.") }
+            ))
+        }
+
+        loginStatus.text = authenticator.currentAccount.asLoginStatus()
+
+        clientCredentials.setOnClickListener {
+            // Obtain logged out credentials which can be used to make logged out requests.
+            authenticator.clientCredentials(vimeoCallback(
+                onSuccess = {
+                    toast("Obtained client credentials.")
+                    loginStatus.text = authenticator.currentAccount.asLoginStatus()
```

Comment:
I added a staff picks fetch and removed the client credentials button. This brings the example more in line with the original, where client credentials are implicitly requested at startup rather than by clicking a button. Now if the user clicks on the get my videos button without being logged in it will set the message to say to log in. However, for staff picks, the consumer will see that they can make the request without being logged in.

Reply:
Overall, I want to keep the example as short as possible. The current example is less than 200 lines of code, and my goal is to keep the `MainActivity` under that. The longer the example and the more complicated it is, the harder it will be for users to grok.